### PR TITLE
[Mobile] Search feedback

### LINKED ISSUE #4605: [mobile ux] Restyle zero search results page
## In a feature branch or master?
Search & Filter feature branch 

## Description
**- As a:** enterprise
**- On page:** my shopfront search
**- I want to be able to do:** have the no results page match the new design of the rest of the shopfront

## In/out of scope for phase 1
Out: Specifying total number of results for a search query
Out: Providing a 'Did you mean...' for common typos/misspellings
Out: Providing a related browse category suggestion like 'Dairy'

## Acceptance Criteria & Tests

Key changes / call outs
- [ ] Present a prominent no results message
- [ ] Present a prominent way to clear the search results (@yukoosawa keep in mind #4602 alongside of this)

**HOW IT LOOKS NOW**
Mobile
Screen Shot 2019-12-18 at 5 10 36 pm

Tablet
Screen Shot 2019-12-18 at 5 10 18 pm

Desktop
Screen Shot 2019-12-18 at 5 10 05 pm

**HOW IT SHOULD LOOK**
**_Mobile_**

Mobile: Shop - 0 Results@1x

**_Tablet_**

Tablet: Shop - 0 Results

**_Desktop_**

Desktop: Shop - 0 Results

Design details are kept in our Zeplin account: https://app.zeplin.io/project/59532764a6c78ff3c534b372/dashboard

Acceptance criteria:
- [ ] Restyled to look like screenshot
- [ ] Clear search function takes user back to full list of products

### LINKED ISSUE #4669: [mobile ux] Restyle the search results functions
## In a feature branch or master?
Search & Filter feature branch 

## Description
**- As a:** shopper
**- On page:** shopfront product list
**- I want to be able to do:** have the search results functions match the designs for the rest of the shopfront 

## Acceptance Criteria & Tests

Key changes / call outs
- [ ] Remove yellow background
- [ ] Update text colours for results line + clear link

**HOW IT LOOKS NOW**
Search results display on all devices
search-mobile

**HOW IT SHOULD LOOK**
**_Mobile_**

Mobile: Shop - Search Results

**_Desktop_**

Desktop: Shop - Search Results

Design details are kept in our Zeplin account: https://app.zeplin.io/project/59532764a6c78ff3c534b372/dashboard

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #4669 
Closes #4605 

Restyles the "no results" message and the "results for..." search query feedback.

#### Release notes

Updating styling on shop search feedback and no results message.

### CODE REVIEW

File: app/assets/stylesheets/darkswarm/_shop-filters.css.scss [hunk 1]
```diff
@@ -107,26 +107,37 @@
...
 
-  span.applied-search {
-    color: $clr-brick;
-  }
+  span {
+    color: $grey-800;
+    font-style: italic;
+
+    &.applied-properties {
+      color: #333;
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-filters.css.scss:148]: #333 is $grey-800 and grey-800 is already the span color. My logic says this is not needed. 
If you find it's needed, it should be $grey-800.

File: app/assets/stylesheets/darkswarm/_shop-filters.css.scss [hunk 2]
```diff
@@ -156,6 +156,14 @@
       font-weight: bold;
       color: $teal-500;
     }
+
+    &.filter-label {
+      float: left;
+    }
+  }
+
+  @media all and (max-width: 1024px) {
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-filters.css.scss:165]: you can use the new breakpoint mixins?

File: app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee [hunk 3]
```diff
@@ -76,12 +76,15 @@ Darkswarm.controller "ProductsCtrl", ($scope, $filter, $rootScope, Products, Ord
...
+    ).join($scope.filtersJoinWord()) if $scope.activeTaxons?
 
   $scope.appliedPropertiesList = ->
     $scope.activeProperties.map( (property_id) ->
       Properties.properties_by_id[property_id].name
-    ).join(" #{t('products_or')} ") if $scope.activeProperties?
+    ).join($scope.filtersJoinWord()) if $scope.activeProperties?
+
+  $scope.filtersJoinWord = ->
+    $sce.trustAsHtml(" <span>#{t('products_or')}</span> ")
```
Comment (luisramos0) [inline app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee:87]: can you explain this magic a little better in the commit message?

File: app/assets/stylesheets/darkswarm/_shop-filters.css.scss (hunk 2 above)
Comment (Matt-Yorkley) [reply app/assets/stylesheets/darkswarm/_shop-filters.css.scss:165]: Ah yes, I rebased to pick up those changes but forgot to update this bit :+1:

File: app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee (hunk 3 above)
Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee:87]: I'll update the message :+1: 

Basically we need to build a really dynamic structure on the page like "Results for: `<search-term>` in `<filter1>` or `<filter2>` or `<filter3>`" and the styling on each of the words needs to be a bit different, specifically the filter terms need to be differently styled to the join words like "in" and "or". We have to generate those bits on the Angular side, but they also have to be wrapped in HTML in a specific way. It was a bit tricky to achieve. 

`$sce.trustAsHtml()` allows us to add some trusted HTML in the strings, in this case the content doesn't involve any user-defined data, so it should be fine.

Comment (Matt-Yorkley) [reply app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee:87]: Lengthy explanation added :+1: :smile:

The diff on the individual commit is relatively clear, when comparing it to what the old line was doing previously.

**Comment by Matt-Yorkley** [comment]
Thanks @luisramos0, I've updated and force-pushed.

File: app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee (hunk 3 above)
Comment (luisramos0) [reply app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee:87]: awesome, thanks for explanning :+1:

File: app/assets/stylesheets/darkswarm/_shop-filters.css.scss (hunk 2 above)
Comment (luisramos0) [reply app/assets/stylesheets/darkswarm/_shop-filters.css.scss]: :+1:

**Comment by Matt-Yorkley** [comment]
> I'd appreciate a screenshot. CSS looks ok but I don't really know what I'm looking at.

Ah sorry, I added a screengrab in the issue, but I forgot to add it in the PR: https://github.com/openfoodfoundation/openfoodnetwork/issues/4669#issuecomment-617102349

Actually I think it's a tiny bit out of date now... a couple of minor styling bits were tweaked since then, but you get the gist.

**Comment by Matt-Yorkley** [comment]
I'll merge these two PRs into the feature branch now, and get them ready for testing :+1:

**Pull request merged**